azimuth_capi/status.py

```diff
@@ -54,6 +54,9 @@ def _reconcile_cluster_phase(cluster):
             LeasePhase.PENDING,
             LeasePhase.STARTING
         }:
+            if cluster.status.phase == ClusterPhase.UNHEALTHY:
+                # preserve the unhealthy state, on timeout
+                return
```

Comment:
Maybe I should add instance.status.timedout_at = "iso date" which triggers the unhealthy state? bit messy as you need to reset that, but it might be better, no sure. Seems a mess to add this in here for each case.